services/movie_session.py

```diff
@@ -0,0 +1,54 @@
+from datetime import datetime
+
+from db.models import MovieSession
+
+
+def create_movie_session(
+        movie_show_time: str,
+        movie_id: int,
+        cinema_hall_id: int
+) -> MovieSession:
+    return MovieSession.objects.create(
+        show_time=movie_show_time,
+        movie_id=movie_id,
+        cinema_hall_id=cinema_hall_id
+    )
+
+
+def get_movies_sessions(session_date: str | None = None) -> MovieSession:
+    if session_date:
+        date_object = datetime.strptime(session_date, "%Y-%m-%d").date()
+
+        return MovieSession.objects.filter(show_time__date=date_object)
+
+    return MovieSession.objects.all()
+
+
+def get_movie_session_by_id(movie_session_id: int) -> MovieSession:
+    return MovieSession.objects.get(id=movie_session_id)
+
+
+def update_movie_session(
+        session_id: int,
+        show_time: str | None = None,
+        movie_id: int | None = None,
+        cinema_hall_id: int | None = None
+) -> MovieSession:
+    try:
+        movie_session = MovieSession.objects.get(id=session_id)
```

Comment:
Try to implement this:

1.     Create an empty dict
2. Check if any of the optional arguments are passed to this function. In case if there are - you will add them to the dict.
3. In the end check if your dict has some values to update and if yes you will call update on queryset, and if there are not any you don't do anything.